server/src/interfaces.ts

```diff
@@ -31,3 +31,36 @@ export interface CustomerPayload {
 export interface CustomerResponse extends CustomerPayload {
   id: number;
 }
+
+/**
+ * ListingPayload Interface that contains the fields of the payload that
+ * would be sent to create a Listing Entity.
+ */
+export interface ListingPayload {
+  merchantId: number;
+  name: string;
+  price: string;
+  oldPrice: string;
+  currency: string;
+  imgUrl: string;
+  description: string;
+  deadline: string;
+  minCommits: number;
+  numCommits: number;
+  numPaid: number;
+  numCompleted: number;
+  listingStatus:
```

Comment:
can we use enum type?

Reply:
Discussed on Chat and decided to stick with using union types, but refactor the listing statuses out to a `ListingStatus` type.